src/midiEditorCommands.cpp

```diff
@@ -18,17 +18,30 @@
 
 using namespace std;
 
+const double DEFAULT_PREVIEW_LENGTH = 0.3;
+
 typedef struct {
 	int channel;
 	int pitch;
 	int velocity;
 	int index;
+	double start;
+	double end;
+	double GetLength() const {
+		double length = end - start;
+		if (length <= 0) {
```

Comment:
It could be negative when providing a higher start than end. That should never happen, but just in case as you say, better safe than sorry. Note that SetTimer doesn't take negative values and therefore the interval of the timer would be uncertain.